source/common/http/conn_manager_impl.h

```diff
@@ -23,6 +23,7 @@
 #include "envoy/tracing/http_tracer.h"
 #include "envoy/upstream/upstream.h"
 
+#include "common/buffer/watermark_buffer.h"
```

Comment:
Q: With the new factory stuff that we have from the other PRs, should we expose watermark directly to conn_manager_impl? Or just use the factory to create a watermark buffer and return an Instance?

Reply:
I mildly prefer doing this for now - the only reason we did the factory is to allow injecting buffers in end to end tests, and it'll complicate the test logic if we use it places we don't need to inject mock buffers.  100% fine if you think it's more consistent though - I can go tweak the tests.

Reply:
Nah I don't care. Was just curious.